reconcile/terraform_repo.py

```diff
@@ -0,0 +1,304 @@
+import logging
+from collections.abc import (
+    Mapping,
+    MutableMapping,
+)
+from typing import (
+    Any,
+    Callable,
+    Optional,
+)
+
+import yaml
+from pydantic import BaseModel
+
+from reconcile import queries
+from reconcile.gql_definitions.terraform_repo.terraform_repo import (
+    TerraformRepoV1,
+    query,
+)
+from reconcile.utils import gql
+from reconcile.utils.defer import defer
+from reconcile.utils.exceptions import ParameterError
+from reconcile.utils.gitlab_api import GitLabApi
+from reconcile.utils.semver_helper import make_semver
+from reconcile.utils.state import (
+    State,
+    init_state,
+)
+
+QONTRACT_INTEGRATION = "terraform-repo"
+QONTRACT_INTEGRATION_VERSION = make_semver(1, 1, 0)
+
+
+class AWSAuthSecret(BaseModel):
+    path: str
+    version: Optional[int]
+
+
+class AWSAccount(BaseModel):
+    name: str
+    uid: str
+    secret: AWSAuthSecret
+
+
+class TFRepo(BaseModel):
+    """This is what the integration outputs and is input for the executor to perform terraform operations"""
+
+    name: str
+    repository: str
+    ref: str
+    project_path: str
+    account: AWSAccount
+    delete: Optional[bool]
+
+
+class ActionPlan(BaseModel):
+    dry_run: bool
+    repos: list[TFRepo]
+
+
+def get_repos(query_func: Callable) -> list[TFRepo]:
```

Comment:
I'm not sure if we should move this to [typed_queries](https://github.com/app-sre/qontract-reconcile/tree/master/reconcile/typed_queries), then integration code only call typed query functions.

Since this is the only usage now, maybe OK to not extract, just not sure about the pattern we should follow.

Reply:
Good question. This is the only integration for the forseeable future that will use this query but later on down the line others may use it.

Reply:
Cool, we can leave it as is for now, extract when reuse is needed.